### DISCUSSION TITLE
Does the TCP KISS Interface support id_callsign?

### DISCUSSION THREAD

**Discussion by iqw1345** [markqvist/Reticulum#463] [General]
I see that the regular KISS interface that uses ports supports id_callsign, and I figured that I would try copying the id_callsign and id_interval stanza from the manual to the TCP KISS Interface. As far as I can tell, it doesn't seem to support it. Would it be in scope for id_callsign to be supported by the TCP KISS Interface?

**Comment by markqvist** [comment]
You're right, one would expect that to work, and it's pretty essential functionality for TCP KISS support. It's definitely in scope for implementation.

**Reply by gretel** [reply]
this would be great to have in a future release. regards

**Comment by gretel** [comment]
this PR fixes the beaconing: https://github.com/markqvist/Reticulum/pull/618
if that gets approved i can work on making this work for the `TCP` interface, too.

**Comment by kenktk** [comment]
Hello,
I would like to poke this as I am testing reticulum on the Vara HF modem right now. The modem complains about the data sent from reticulum not having a callsign. 

<img width="238" height="128" alt="image" src="https://github.com/user-attachments/assets/2abf9dde-dc38-4e77-a3fe-db3f73a2b0e5" />

The interface is described as:
`  [[KISS-TCP-VARA]]
    type = TCPClientInterface
    interface_enabled = true
    target_host = 127.0.0.1
    target_port = 8100
    kiss_framing = True`

adding id_callsign does not fix the issue it seems.  Here is the link to the Vara KISS manual (it is also attached):
[https://mega.nz/file/eCxBFZzA#F6mfrEtQ0UCh7dxxKinwgU0livyB_87QI2delWbYn_U](url)
[VARA KISS Interface.pdf](https://github.com/user-attachments/files/21192308/VARA.KISS.Interface.pdf)

**Reply by RFnexus** [reply]
From the manual it looks like VARA is expecting a field for determining the framing. My guess is it's currently looking for AX.25 framed packets when we want generic data. 
<img width="884" height="311" alt="image" src="https://github.com/user-attachments/assets/26a8209e-1f06-4e0f-9d09-91d77e219568" />

The `id_callsign` option is used for beaconing in the `KISSInterface` and isn't a supported option yet for the `TCPClientInterface`  (wouldn't help here)

**Reply by RFnexus** [reply]
Were you able to get this working? If you want I can copy the TCPClientInterface, modify a few things in the KISS framing and post it here for use as a CustomInterface to get VARA and Reticulum communicating
